tome-app/src/art-pages/ArtworkDetails.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from "react";
+import { collection, getDocs, doc, getDoc, onSnapshot } from "firebase/firestore";
+import { useParams, Link } from 'react-router-dom';
+import { auth, db } from '../FirebaseConfig';
+import TopNav from "../TopNav";
+
+export const handleDelete = async (id, user, setArtworks) => {
+    try {
+        await deleteDoc(doc(db, 'accounts', user.uid, 'artworks', id));
+        setArtworks(prevArtworks => prevArtworks.filter(artwork => artwork.id !== id));
+        console.log("Deleted artwork:", id);
+    } catch (error) {
+        console.error("Error with the delete:", error)
+    }
+}
+
+function ArtworkDetails() {
+    const user = auth.currentUser;
+    const { id } = useParams();
+    const [artwork, setArtwork] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [hasError, setHasError] = useState(false);
+
+    //show details
+    useEffect(() => {
+
+        if (!user) {
+            console.log("No authenticated user. Skipping Firestore request.");
+            return; // Exit early if user is not authenticated
+        }
+
+        const getData = async () => {
+
+
+
+            try {
+                const docRef = doc(db, "accounts", user.uid, "artworks", id);
+                const docSnap = await getDoc(docRef);
+                if (docSnap.exists()) {
+                    setArtwork(docSnap.data())
+                } else {
+                    setHasError(true);
+                }
+
+                if (docSnap.exists()) {
+                    setArtwork(docSnap.data());
+                } else {
+                    console.log("Didn't find that artwork");
+                    setHasError(true);
+                }
+
+            } catch (error) {
+                console.error("Updated error fetching document:", error);
+                setHasError(true);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        getData();
+    }, [id, user])
+
+    console.log(artwork);
+
+    return (
+        <>
+            {isLoading ? (
+                <p>Loading...</p>
+            ) : hasError ? (
+                <p>There was a problem loading the artwork.</p>
+            ) : artwork ? (
```

Comment:
Nested ternaries like this are definitely a pro move. They're not my preference as they can be hard to read, but this is nice and clean.